### PR TITLE
HLRC: Adding Update datafeed API

### PR DESCRIPTION
**Pull request by benwtrent**
HLRC: ML Adds the Update Datafeed API

Relates to #29827

### CODE REVIEW

**Comment by benwtrent** [comment]
Pinging @elastic/ml-core

File: client/rest-high-level/src/main/java/org/elasticsearch/client/MLRequestConverters.java [hunk 1]
```diff
@@ -209,6 +210,19 @@ static Request putDatafeed(PutDatafeedRequest putDatafeedRequest) throws IOExcep
         return request;
     }
 
+    static Request updateDatafeed(UpdateDatafeedRequest updateDatafeedRequest) throws IOException {
+        String endpoint = new EndpointBuilder()
+            .addPathPart("").addPathPartAsIs("_xpack")
```
Comment (dimitris-athanasiou) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/MLRequestConverters.java]: The first empty part seems redundant

File: client/rest-high-level/src/main/java/org/elasticsearch/client/MachineLearningClient.java [hunk 2]
```diff
@@ -494,6 +495,46 @@ public void putDatafeedAsync(PutDatafeedRequest request, RequestOptions options,
                 Collections.emptySet());
     }
 
+    /**
+     * Updated a Machine Learning Datafeed
```
Comment (dimitris-athanasiou) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/MachineLearningClient.java]: typo: Updated -> Updates

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/UpdateDatafeedRequest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import org.elasticsearch.client.ml.datafeed.DatafeedUpdate;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Updates a {@link org.elasticsearch.client.ml.datafeed.DatafeedConfig} with the passed {@link DatafeedUpdate}
```
Comment (dimitris-athanasiou) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/UpdateDatafeedRequest.java]: This comment implies this class is the action. Reword to suit it is a request?

File: client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java [hunk 4]
```diff
@@ -357,6 +359,33 @@ public void testPutDatafeed() throws Exception {
...
+        assertThat(createdDatafeed.getId(), equalTo(datafeedId));
+        assertThat(createdDatafeed.getIndices(), equalTo(datafeedConfig.getIndices()));
+
+        DatafeedUpdate datafeedUpdate = DatafeedUpdate.builder(datafeedId).setIndices("some_other_data_index").setScrollSize(10).build();
+
+        response = execute(new UpdateDatafeedRequest(datafeedUpdate),
+            machineLearningClient::updateDatafeed,
+            machineLearningClient::updateDatafeedAsync);
+
+        DatafeedConfig updatedDatafeed= response.getResponse();
```
Comment (dimitris-athanasiou) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java]: nit: missing space before equals

File: docs/java-rest/high-level/ml/update-datafeed.asciidoc [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+A +{request}+ requires the following argument:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-request]
+--------------------------------------------------
+<1> The updated configuration of the {ml} datafeed
+
+[id="{upid}-{api}-config"]
+==== Updated Datafeed Configuration
```
Comment (dimitris-athanasiou) [inline docs/java-rest/high-level/ml/update-datafeed.asciidoc]: This title/section might be unnecessary. Check `update-job` documentation. We could have a similar structure here unless you dislike that one.

**Review by dimitris-athanasiou — COMMENTED** [review]
Looks good. Left a few minor comments.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/MLRequestConverters.java (hunk 1 above)
Comment (benwtrent) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/MLRequestConverters.java]: definitely :D copy paste error

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/UpdateDatafeedRequest.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+import org.elasticsearch.client.ml.datafeed.DatafeedUpdate;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Requests an updates to a {@link org.elasticsearch.client.ml.datafeed.DatafeedConfig} with the passed {@link DatafeedUpdate}
```
Comment (dimitris-athanasiou) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/UpdateDatafeedRequest.java]: nit: updates -> update

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM apart from a little typo

**Pull request merged**